aspnetcore/includes/mvc-intro/search3.md

```diff
@@ -55,4 +55,10 @@ Update `Index.cshtml` as follows:
 
 [!code-HTML[Main](../../tutorials/first-mvc-app/start-mvc/sample/MvcMovie/Views/Movies/IndexFormGenreNoRating.cshtml?highlight=1,15,16,17,28,31,34,37,43)]
 
-Test the app by searching by genre, by movie title, and by both.
\ No newline at end of file
+Examine the lambda expression used in the following `HTML Helper`:
+
+ `@Html.DisplayNameFor(model => model.movies[0].Title)`
+ 
+The lambda expression in the preceeding code gets the meta-data for the HTML Helper from the movie model, not from an instance of the movie array. That's why, when the model is empty (that is, there is no element at `model.movies[0]`), you don't get an access violation. The lambda expression is not accessing data. 
```

Comment:
This isn't horrible but it's phrased in terms of the lambda expression doing something. When the expression is actually evaluated, it will be run against the actual model. However, `Html.DisplayNameFor()` just "examines" (perhaps "inspects", perhaps something else -- up to you) the lambda expression to determine the appropriate name _even_ when `model`, `model.movies`, or `model.movies[0]` are `null` or empty.

Could also be phrased in terms of what MVC does with these lambda expressions. The general point is MVC deals with the expression metadata until the very last moment. And, for something like `DisplayNameFor()` that just needs a name or other metadata, the lambda is _never_ "executed", "evaluated", or whatever you think is best. 

FYI the MVC doc comments related to lambda expressions tend to use "evaluated".